### ISSUE TITLE
Cart block - final styling, mobile and theme support review

### ISSUE THREAD

**Issue by haszari** [woocommerce/woocommerce-gutenberg-products-block#1539]
This issue covers styling fixes to ensure that the cart looks and behaves as intended by the design. Much of this should be included in PRs for specific areas (e.g. quantity picker #1521, remove item from cart #1534) – this issue is to ensure nothing falls through the cracks. Examples: line item price column text styling; text sizes and colours overall.

The cart block should work well with Storefront theme, and ideally not cause issues with other themes. This issue also includes testing and fixes to achieve that. 

This issue also covers any remaining mobile / responsive fixes.

For example, the design uses admin font (`SF UI` i.e. system font), and Storefront uses `Source Sans Pro`, so we should at least review the implemented design (when work is nearing completion). cc @garymurray 

Here's a pic of the current state of the cart (work in progress) vs the design as a snapshot:

![Screen Shot 2020-01-10 at 2.19.01 PM.png](https://images.zenhubusercontent.com/5ddb3cb39529f10001e618e5/db4eb9d2-13bc-4751-a6fa-18cb18771d16)

![Screen Shot 2020-01-10 at 2.21.07 PM.png](https://images.zenhubusercontent.com/5ddb3cb39529f10001e618e5/84f7f24a-3d27-4c10-a158-067754ac874d)

**Comment by haszari** [comment]
The current font sizes & layout are not working well in the cramped space in the editor: 

<img width="800" alt="Screen Shot 2020-01-13 at 11 03 04 AM" src="https://user-images.githubusercontent.com/4167300/72226277-60c46f00-35f4-11ea-8bdb-eac256a023da.png">

(Note wrapped column header and discount badge.)

**Comment by garymurray** [comment]
@haszari is the screenshot above a site with Storefront that has is using the sidebar template for the cart page?

**Comment by haszari** [comment]
> is the screenshot above a site with Storefront that has is using the sidebar template for the cart page?

Yes, here's a screenshot of front end using `Full width` template for comparison :)

<img width="1104" alt="Screen Shot 2020-01-14 at 9 03 55 AM" src="https://user-images.githubusercontent.com/4167300/72287950-db50c580-36ac-11ea-9a03-b0c22f7cba02.png">

**Comment by garymurray** [comment]
Thanks - it looks like Storefront will need some styling tweaks to accommodate the new cart and checkout blocks.

**Comment by haszari** [comment]
We've accumulated a few issues on mobile.

- Low stock warning badge needs margins.
- Line item subtotals / prices and remove button are off screen.
- Totals card wraps and is too small - i.e. we need to implement the mobile design.
- Coupon code field inline label wraps (may go away when we fix totals card).

<img width="1111" alt="Screen Shot 2020-02-27 at 1 39 25 PM" src="https://user-images.githubusercontent.com/4167300/75401375-d8f9b080-5966-11ea-86ef-1cf22059decc.png">

Note - some of these may be regressions, there has been some work on mobile support as we work on each area.

**Comment by haszari** [comment]
Added some issues for some specific things we can fix right now:

 - [line items area styling issues](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1870)
 - [line low stock badge <=> product padding/margin](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1871)
 - [totals card on mobile](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1872)
 - [sticky checkout button](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1873)

**Comment by haszari** [comment]
Storefront allows various colors to to be changed in customizer:

<img width="335" alt="Screen Shot 2020-03-09 at 4 00 13 PM" src="https://user-images.githubusercontent.com/4167300/76179246-14299880-621f-11ea-94d6-2d4d626ee371.png">

We're using the text color inconsistently in the cart block:

<img width="1365" alt="Screen Shot 2020-03-09 at 3 58 26 PM" src="https://user-images.githubusercontent.com/4167300/76179225-007e3200-621f-11ea-8c9d-fdae3647f198.png">

**Comment by Aljullu** [comment]
@garymurray I noticed in the _Cart_ and _Checkout_ blocks product names should be `#40464D` and not underlined according to the designs. But that's quite different from the styles the product name has in the _All Products_ block (link color + underline). Would make it sense unifying both styles? Or the difference is there in purpose?

_Cart_ block designs:
![imatge](https://user-images.githubusercontent.com/3616980/79878411-ae1f5c00-83ed-11ea-93d5-c4f5f02b441d.png)

_All Products_ block:
![imatge](https://user-images.githubusercontent.com/3616980/79878476-be373b80-83ed-11ea-95c5-d07f28519aed.png)

For reference, in the shop page they have no underline and inherit the heading color.

**Comment by garymurray** [comment]
I'd go with unifying around no underline. 

Regarding the colour - what does the current shortcode use for the product names - where is it pulling the colour from?

**Comment by Aljullu** [comment]
> Regarding the colour - what does the current shortcode use for the product names - where is it pulling the colour from?

@garymurray By default, it's rendered as a normal link, like in _All Products_ block (purple with underline). But Storefront overwrites those styles removing the underline and changing its color to the heading color:

![imatge](https://user-images.githubusercontent.com/3616980/80090725-3c1b5400-8560-11ea-875e-b771671fe343.png)

That's a tricky issue, I don't think we should apply the _Heading color_ to those links. We probably can't undo that for shortcodes without breaking sites, but I think new blocks should instead use the _Text color_ or the _Link color_, ideally.

**tl;dr**

I think we need to answer these questions:
- Should blocks enforce a specific styles for Product name links (color & underline)? I would say no, that should be left to the theme.
- If the answer is no:
  - Should Storefront remove the underline of product name links?
  - What color should Storefront use for product name links? Shortcodes use the _Heading color_ but that seems wrong to me. I think for new blocks we should either use the default _Text_ or _Link color_.

**Comment by garymurray** [comment]
> Should blocks enforce a specific styles for Product name links (color & underline)? I would say no, that should be left to the theme.

I think blocks should use the text color by default as even though it is a link - you don't really want it to stand out and make people click it in the cart unless they really need to, and the theme can override that if it wants.

> Should Storefront remove the underline of product name links?

I think this is a styling thing specific to the theme, and worth leaving as is for now.

> What color should Storefront use for product name links? Shortcodes use the Heading color but that seems wrong to me. I think for new blocks we should either use the default Text or Link color.

I think it should be using the link color here as it is a link as it a link you'd want people to potentially click. 

Any input here @LevinMedia ?
